### ISSUE TITLE
JWT_Auth Section

### ISSUE THREAD

**Issue by NicoHinderling** [box/box-python-sdk#64]
I'm currently trying out your JWT_Auth component of the SDK, and I've noticed a couple things. 

1.) Someone should probably correct the fact that there is no "enterprise_token", but rather should be "enterprise_id".
2.) You should probably add "rsa_private_key_passphrase" in there and remove the default value. https://box-content.readme.io/v2.0/docs/box-developer-edition provides a nice script to make a .PEM, yet by default will require a passphrase. It only makes sense to either make the Dev docs not create a passphrase by default, or make a passphrase a component of this SDK.

3.) This is less important, but consider adding references to where to find the enterprise_id. I was searching for 30 minutes before calling support and getting pointed to account settings in the admin console of my "enterprise". Definitely not necessary, but would be a nice thing to have for first time users like myself.

**Comment by Jeff-Meadows** [comment]
Hi Nico - thanks for the report.

1) Good catch. I'm not sure why the SDK calls the enterprise_id and enterprise_token - we should fix that.
2) The JWT auth constructor does accept a passphrase, but the default value is None which works for key files with no passphrase. I agree this is confusing since the examples in the README don't include using a passphrase. I'd rather not make it a required parameter, but I think the examples should illustrate using a passphrase.
3) Good feedback. I think this is changing quickly until Developer Edition reaches GA (which should be relatively soon). I'll go ahead and add a note about where to find it currently.

**Comment by NicoHinderling** [comment]
Thanks for the quick response, Jeff. I think you're probably right about #2 now that I think about it. I'm still encountering some issues but I'll shoot in a more formal issue if it persists. Thanks again!

**Comment by NicoHinderling** [comment]
Hmm well I hope you don't mind, but I've been trying to make a call with the JWT_Auth, and I keep getting this response:

```
<boxsdk.network.default_network.DefaultNetworkResponse object at 0x109d95d10>
Traceback (most recent call last):
  File "jwt_client.py", line 18, in <module>
    access_token = auth.authenticate_instance()
  File "/Users/nico/.virtualenvs/box/lib/python2.7/site-packages/boxsdk/auth/jwt_auth.py", line 173, in authenticate_instance
    return self._auth_with_jwt(self._enterprise_token, 'enterprise')
  File "/Users/nico/.virtualenvs/box/lib/python2.7/site-packages/boxsdk/auth/jwt_auth.py", line 146, in _auth_with_jwt
    return self.send_token_request(data, access_token=None, expect_refresh_token=False)[0]
  File "/Users/nico/.virtualenvs/box/lib/python2.7/site-packages/boxsdk/auth/oauth2.py", line 226, in send_token_request
    raise BoxOAuthException(network_response.status_code, network_response.content, url, 'POST')
boxsdk.exception.BoxOAuthException: 
Message: {"error":"invalid_grant","error_description":"Signature verification failed"}
Status: 400
URL: https://api.box.com/oauth2/token
Method: POST
```

Here is the code: 
```
from boxsdk import JWTAuth


def store_tokens(access_token, refresh_token):
    print access_token
    print "_!_!_!_!_!_!_!_!_!_!_!_!_!_!_!_!_"
    print refresh_token


auth = JWTAuth(
    client_id='bovr10j9pyivk4i3pne82fjs8x2imgrg',
    client_secret='[redacted]',
    enterprise_id='685907',
    rsa_private_key_file_sys_path='/Users/nico/Code/ds_box/Keys/private_key.pem',
    rsa_private_key_passphrase='[redacted]'
)

access_token = auth.authenticate_instance()

from boxsdk import Client

client = Client(auth)
```

Any idea what could be the issue?

**Comment by NicoHinderling** [comment]
Nevermind... I just realized it was an account issue. Thanks for all of this. I'll let you close this whenever you get around to making the small readme edit. Thanks again.

**Comment by NicoHinderling** [comment]
I actually have realized now that I can't find where to upload my public key (for my keypair...). 

```
"You will then provide Box with the Public Key necessary to verify the JWT assertion signed with your application's Private Key. During the closed Beta period, we will provide a form for entering the Public Key. Following the closed Beta, you will enter the Public Key in the services page for your application."
```

I'm not sure where this is referring to...

**Comment by Jeff-Meadows** [comment]
I'm not sure about that - please email developer-edition@box.com for access to the form.

**Comment by Frencil** [comment]
@NicoHinderling:

> I actually have realized now that I can't find where to upload my public key (for my keypair...).

FYI I've been trying to set up JWT Auth too and just today I've noticed that the services page for the application (where I set its name, get a developer token, etc.) now has an area to upload public keys. This was after a few days of trying (and failing?) to set up my box developer account (silent failure of their registration form as far as I could tell).

At one point, after contacting Box developer support, I did receive an invitation to a Google Form that included a field to upload my public key, which I completed, but afterward learned that this is a feature only available to individual users and not enterprises (I was trying to register my Box account that's tied to a large enterprise account).

So check now, maybe you have it too.
